schematics/utils/routing.ts

```diff
@@ -0,0 +1,321 @@
+/**
+ * @license
+ * Copyright Akveo. All Rights Reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import * as ts from 'typescript';
+import { basename, dirname, join, normalize, NormalizedSep, Path } from '@angular-devkit/core';
+import { SchematicsException, Tree } from '@angular-devkit/schematics';
+import { getSourceFile } from '@angular/cdk/schematics';
+import { getDecoratorMetadata, insertImport } from '@schematics/angular/utility/ast-utils';
+import { buildRelativePath } from '@schematics/angular/utility/find-module';
+import {
+  addArrayElement,
+  addObjectProperty,
+  applyInsertChange,
+  findDeclarationByIdentifier,
+  importPath,
+  isBasePlaygroundModule,
+  isLayoutPath,
+  removeBasePath,
+} from '../utils';
+
+const LAYOUT_COMPONENT_CLASS = 'PlaygroundLayoutComponent';
+const NO_LAYOUT_COMPONENT_CLASS = 'PlaygroundBaseComponent';
+
+export function findRoutesArray(tree: Tree, modulePath: Path): ts.ArrayLiteralExpression {
```

Comment:
Let's add some comments here, it's quite hard to understand

Reply:
Also, don't you think this function is freaking big? It doesn't fit in my monitor 😄 Let's split it somehow